keps/sig-windows/2802-identify-windows-pods-apiserver-admission/README.md

```diff
@@ -0,0 +1,401 @@
+
+The runtimeclass is a higher level abstraction which gets translated again to nodeSelectors+tolerations. While the 
+nodeSelector with reserved OS label is good enough, it has following shortcomings:
+
+- It can be viewed as an side-effect of scheduling constraints 
+and need not necessarily express the user intention fully.
+- The pod may target Windows node but pod OS may be linux or windows
+as Linux Containers on Windows(LCOW) can be supported in future.
+
+So, we propose to add a new field to the pod spec called 
+os to identify the OS of the containers specified in the pod. There is no default
+value for this `OS` field or the `Name` field in `OS` struct.
+
+```go
+type PodSpec struct {
+  // If specified, indicates the type of OS on which pod intends
+  // to run.  This field is alpha-level and is only honored by servers that 
+  // enable the IdentifyPodOS feature. This is used to help identifying the
+  // OS authoritatively during the API server admission.
+  // +optional
+  OS *OS
+}
+
+// OS has information on the type of OS. 
+// We're making this a struct for possible future expansion.
+type OS struct {
+  // Name of the OS. The values supported are available at:
+  // https://github.com/opencontainers/runtime-spec/blob/master/config.md#platform-specific-configuration
+  Name string
+}
+
+```
+
+With the above change, all the admission plugins which validate or mutate the pod can 
+identify the pod's OS authoritatively and can act accordingly. As of now, PodSecurityAdmission
+plugin is the only admission plugin can leverage this field on the pod spec. In addition, API validators like ValidatePod and ValidatePodUpdate should be modified. In future, we can
+have a validating admission plugin for Windows pods as Linux and Windows host capabilities are
+different and not all Kubernetes features are supported on Windows host.
+
+
+### User Stories (Optional)
+
+
+#### Story 1
+
+As a Kubernetes cluster administrator, I want appropriate security contexts to be 
+applied to my Windows Pods along with Linux pods
+
+#### Story 2
+
+As a Kubernetes cluster administrator, I want to use my own admission webhook for pods
+based on the OS it intends to run on.
+
+### Notes/Constraints/Caveats (Optional)
+
+Since the OS field is optional in the pod spec, we can expect the users to omit this field
+when submitting pod to API server. In this scenario, we expect admission plugins to treat the pod
+the way it is being treated now.
+
+### Risks and Mitigations
+
+The primary risk is a bug in the implementation of the admission plugins that validate or mutate
+based on the OS field in the pod spec. The best mitigation for that scenario is unit testing 
+when this featuregate is enabled and disabled.
+Additionally, there may be some end-user confusion on the functional consequences of setting the new OS field, given that it is optional.
+
+## Design Details
```

Comment:
need sections detailing changes to kube-apiserver and pod security standards (I'd suggest the list from https://github.com/kubernetes/enhancements/pull/2803#discussion_r695196825)

For each accepted OS value:

* Pod spec API validation will be adjusted to ensure values are not set for OS-specific fields that are irrelevant to the Pod's OS.
* Unit tests will be added to ensure new fields in the pod spec are classified as OS-specific or not (and which OSes they are allowed for)

* Pod Security Standards will be reviewed and updated to indicate which Pod OSes they apply to
* The `restricted` Pod Security Standard will be reviewed to see if there are OS-specific requirements that should be added
* The PodSecurity admission implementation will be updated to skip checks which do not apply to the Pod's OS.

Reply:
Included the changes from comments in separate sections. I also added some text around e2e [below](https://github.com/kubernetes/enhancements/pull/2803/files#diff-0bd86b6b4704c4e10a453c7f15e74e31aacb53891087cf069ae11b8bf5293a30R154)

Reply:
+1 for e2e tests demonstrating  functionality on windows nodes